### ISSUE TITLE
VPN asks to enter password to connect on macOS

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#41209]
### Description

VPN asks to enter password to connect on macOS

### Steps to reproduce

1. Open VPN panel to connect, shows Unable to load credentials
2. Set automatically connect to VPN setting enabled
3. Prompt shows up to enter password
4. Doesn't connect to VPN 
5. Can't connect to VPN until credentials load up

### Actual result

![image](https://github.com/user-attachments/assets/fa80839d-3bd1-4723-b863-efc53745acf9)

### Expected result

Should be able to connect to VPN without issues

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave | 1.72.26 Chromium: 129.0.6668.59 (Official Build) nightly (arm64)
-- | --
Revision | 4580158f4966388aa15f701b5fa01cbe22846cc5
OS | macOS Version 15.0 (Build 24A335)

### Channel information

- [X] release (stable)
- [X] beta
- [X] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

cc: @simonhong

**Comment by bsclifton** [comment]
@srirambv you can connect in browser, but just not from the OS, right?

**Comment by srirambv** [comment]
@bsclifton No, it doesn't connect either from OS network connection or from the VPN panel

**Comment by bsclifton** [comment]
It seems like disabling the auto-reconnect helps to fix the issue

**Comment by iambrianfung** [comment]
i’m impacted by this same issue. happy to help diagnose/troubleshoot

**Comment by simonhong** [comment]
Similar issue from web searching. https://www.reddit.com/r/ProtonVPN/comments/wei01h/macos_asking_for_my_vpn_password_proton_password/

**Comment by simonhong** [comment]
> i’m impacted by this same issue. happy to help diagnose/troubleshoot

Do you use multiple vpn client?
and did you try to connect another vpn client while brave vpn is trying to connect in background(because of ondemand)?

**Comment by iambrianfung** [comment]
@simonhong only the corporate wireguard config.

**Comment by simonhong** [comment]
> [@simonhong](https://github.com/simonhong) only the corporate wireguard config.

Thanks, can you remember corporate wireguard was connected also when that happened?

**Comment by iambrianfung** [comment]
yes, i believe it was successfully connected.

i disabled auto-reconnect and haven't had the issue. i can turn it back on if it's helpful for you but it's extremely disruptive to my ability to use my machine since it affects internet connection including slack, etc.

**Comment by bsclifton** [comment]
Quick note - this has also happened to @ShivanKaul. In his case, disabling auto-reconnect did not solve the issue. He had to delete the VPN config and then reconnect.

**Comment by arthuredelstein** [comment]
Happened to me too

**Comment by Brave-Matt** [comment]
Another user reporting this behavior:
https://community.brave.com/t/vpn-wireguard-messing-up-my-computer/602799/2

**Comment by simonhong** [comment]
When this happens, browser state is `Unable to load credentials`

![Image](https://github.com/user-attachments/assets/5182773b-b58c-4b7f-aedd-a2e99c2d4efa)

**Comment by SprayCanMan** [comment]
> Seeing 2-3 users a week reach out about this issue. Disabling `Reconnect Automatically` seems to resolve it for them.

Where do you see this configuration setting?  I'm unable to find it within the VPN settings.

**Comment by simonhong** [comment]
Asked to apple community support to get any clues. - https://discussions.apple.com/thread/256047267

**Comment by simonhong** [comment]
> > Seeing 2-3 users a week reach out about this issue. Disabling `Reconnect Automatically` seems to resolve it for them.
> 
> Where do you see this configuration setting? I'm unable to find it within the VPN settings. Also, I have it already disabled in Brave itself:
> 

That toggle is the only place to turn it off. Do you see this bug after off?

**Comment by LaurenWags** [comment]
just encountered this on macOS x64 15.x, had to delete the VPN config in order to get that "VPN Connection" modal to go away. toggling the auto reconnect setting didn't help at all.

**Comment by srirambv** [comment]
Started encountring this issue again. Happened everytime the machine display gets turned off or system goes to idle state. Also noticing that the reconnection works fine but no internet. Have to disconnect/reconnect a few times for it to work

**Comment by ShivanKaul** [comment]
Another report on X having this issue.

**Comment by alekspop26** [comment]
Will be fixed in Architecture 2.0; let's keep the ticket open for tracking purposes.
